src/main/java/seedu/address/model/property/status/Completion.java

```diff
@@ -0,0 +1,26 @@
+package seedu.address.model.property.status;
+
+public class Completion implements Status {
+    public static final String TOSTRING_MESSAGE = "Completed sale at ";
+
+    private final Offer amount;
+
+    Completion(Offer amount) {
+        this.amount = amount;
+    }
+
+    @Override
+    public Completion next() {
+        return this;
+    }
+
```

Comment:
It seems weird to me that `next()` for `Completion` returns a `Completion`?

Reply:
yeah, cos it has to implement the status interface, and completion is like the end point. Is there a better way to resolve this issue?

Reply:
> It seems weird to me that `next()` for `Completion` returns a `Completion`?

Since it implements `Status` it needs to have a `next()`, and `Completion` shouldn't lead to any new statuses anyway, so I guess it's fine to return itself? Creating new objects at a terminal step may lead to some undesirable side effects. By checking if `.next()` returns the same object it can even be used to check if something is calling `.next()` on a `Completion`.

Reply:
Would it be better if the `.next()` method is never called for `Completion`? But I do understand that it has to implement the method in this scenario. So I guess its fine.

But perhaps one other suggestion is to show the user that it is already completed or something if user pass in a command to update a completed property to the next stage?

Reply:
added a check to raise an exception if proceed is used on a property with a Completion status